torch/_inductor/codegen/cpp_gemm_template.py

```diff
@@ -121,9 +119,15 @@
                         {{ micro_gemm.codegen_call(kernel, tile_X, tile_W, acc, accum=True)|indent(24, false) }}
                     }
                 }
-                {%- set tile_Y = kernel.slice_nd(Y_2d, [("m_start", "m_end"), ("n_start", "n_start + N0")]) %}
+                {%- if N == PADDED_N %}
+                    {%- set tile_Y = kernel.slice_nd(Y_2d, [("m_start", "m_end"), ("n_start", "n_start + N0")]) %}
```

Comment:
Can this line also be `{%- set tile_Y = kernel.slice_nd(Y_2d, [("m_start", "m_end"), ("n_start", "n_end")]) %}`, so we can share same line as implementation of `padded_n`?

Reply:
For cases without padding, I don't want the expressions depend on `n_end` which is runtime computed but on the constant `N0`. That's why I used the conditionals...